File: vue.jsonforms/src/entities/firm-details/Employee.ts
import { ContactNumber } from "../ContactNumber";
import { ProductType } from "@/entities/org-structure/ProductType";
import Role from "@/entities/org-structure/Role";
import { IEmployeeTask } from "@/entities/org-structure/IEmployeeTask";
import {CustomerBase} from "@/entities/CustomerBase";

export class Employee extends CustomerBase {
  public companyNo: string = "";
  public contactNumber: ContactNumber = new ContactNumber();
  public isRoot: boolean = false;

  public status?: string = "";
  public customerId?: string;
  public title?: string;
  public lineManager?: Employee;
  public primaryRole?: Role;
  public otherRoles?: Role[] = [];
  public productType?: ProductType[] = [];
  public img_id?: string = "";
  public employmentStatus?: string = "Onboarding";
  public tasks?: IEmployeeTask[] = [];
}

File: CSharp.Functions/Common/Entities/FirmDetails/OrganizationalStructure/Employee.cs
namespace Common.Entities
{
    public class Employee: CustomerBase
    {
        public string? CustomerId { get; set; }
        public string? CompanyNo { get; set; }
        public string? Title { get; set; }
        public Employee? LineManager { get; set; }
        public Role? PrimaryRole { get; set; }
        public List<Role> OtherRoles { get; set; } = new List<Role>();
        public List<ProductType> ProductType { get; set; } = new();
        public string? EmploymentStatus { get; set; }
        public bool IsRoot { get; set; }
        public string? ImgId { get; set; }
        public List<EmployeeTask> Tasks { get; set; } = new();
        public string OnboardingType { get; set; } = OnboardingTypes.Employee.ToString();
    }
}